packager/mpd/base/mpd_utils.cc

```diff
@@ -354,6 +360,22 @@ Element GenerateCencPsshElement(const std::string& pssh) {
   return cenc_pssh;
 }
 
+// Extract MS PlayReady Object from given PSSH
+// and encode it in base64
+Element GenerateMsprProElement(const std::string& pssh) {
+  std::unique_ptr<media::PsshBoxBuilder> b =
+    media::PsshBoxBuilder::ParseFromBox((const uint8_t*)pssh.data(), pssh.size());
+
+  const std::vector<uint8_t> *p_pssh = &b->pssh_data();
+  std::string base64_encoded_mspr;
+  base::Base64Encode(base::StringPiece((const char*)p_pssh->data(), p_pssh->size()),
```

Comment:
Similar to above. Use C++ style cast reinterpret_cast.

Reply:
Done